module/documents/actor.mjs

```diff
@@ -115,84 +143,267 @@ export default class ActorEd extends Actor {
-      devotion: { pointsUsed: ability.system.devotionRequired ? 1: 0, pointsRequired: ability.system.devotionRequired, available: this.system.devotion.value, step: this.system.devotion.step },
-      chatFlavor: ability.name + " Test",
+    const difficultyFinal = { base: difficulty }
+    const devotionRequired = ability.system.devotionRequired ? true : false;
+    const strain = { base: ability.system.strain };
+    const chatFlavor = game.i18n.format( "ED.Chat.Flavor.rollAbility", {
+      sourceActor: this.name,
+      ability: ability.name,
+      step: abilityFinalStep
     } );
+    const edRollOptions = await this.createEdRollOptions( "action", "ability", strain, difficultyFinal, abilityFinalStep, devotionRequired, chatFlavor );
     const roll = await RollPrompt.waitPrompt( edRollOptions, options );
     this.#processRoll( roll );
   }
 
-   /**
+  /**
    * @summary                     Equipment rolls are a subset of Action test resembling non-attack actions like Talents, skills etc.
    * @description                 Roll an Equipment. use {@link RollPrompt} for further input data.
    * @param {ItemEd} equipment    Equipment must be of type EquipmentTemplate & TargetingTemplate
    * @param {object} options      Any additional options for the {@link EdRoll}.
    */
-   async rollEquipment( equipment, options = {} ) {
+  async rollEquipment( equipment, options = {} ) {
     const arbitraryStep = equipment.system.usableItem.arbitraryStep
-    const difficulty = equipment.system.getDifficulty();
-    if ( !difficulty ) {
-      ui.notifications.error( game.i18n.localize( "X.ability is not part of Targeting Template, please call your Administrator!" ) );
+    const difficulty = await equipment.system.getDifficulty();
+    if ( difficulty === undefined || difficulty === null ) {
+      ui.notifications.error( "ability is not part of Targeting Template, please call your Administrator!" );
       return;
     }
-    const edRollOptions = new EdRollOptions( {
-      testType: "action",
-      step: { base: arbitraryStep },
-      target: { base: difficulty },
-      karma: { pointsUsed: this.system.karma.useAlways ? 1 : 0, available: this.system.karma.value, step: this.system.karma.step },
-      devotion: { available: this.system.devotion.value, step: this.system.devotion.step },
-      chatFlavor: equipment.name + " Equipment Test",
+    const difficultyFinal = { base: difficulty }
+    const chatFlavor = game.i18n.format( "ED.Chat.Flavor.rollEquipment", {
+      sourceActor: this.name,
+      equipment: equipment.name,
+      step: arbitraryStep
     } );
+    const edRollOptions = await this.createEdRollOptions( "action", "equipment", 0, difficultyFinal, arbitraryStep, false, chatFlavor );
     const roll = await RollPrompt.waitPrompt( edRollOptions, options );
     this.#processRoll( roll );
   }
 
+  async takeDamageManual( options = {} ) {
```

Comment:
Why is this necessary? For the prompt?
That can be incorporated in the `takeDamage` function. Or rather: make it like the other prompts: call it from the listener and pass the data to the `takeDamage` function

Reply:
how do I avoid calling the prompt if i incorporate it into takeDamage? that was my reason to exclude it

Reply:
takeDamageManual in den BaseSheet am listener und nach prompt dann ganz normal takeDamage im actor aufrufen.